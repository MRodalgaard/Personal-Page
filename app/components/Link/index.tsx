import { MouseEvent, ReactNode } from 'react';
import styled from 'styled-components';
import { AnalyticsEvent, logEvent } from 'utils/analytics';

const StyledLink = styled.a`
  color: ${({ color, theme }) => (color ? color : theme.primary)};
  text-decoration: underline;
  cursor: pointer;
  transition: color 0.5s ease;

  &:hover {
    color: ${({ theme }) => theme.secondary};
  }
`;

export default function Link({
  children,
  className,
  color,
  href,
  ariaLabel,
  analyticsEvent,
  onClick,
}: {
  children: ReactNode;
  className?: any;
  color?: string;
  href?: string;
  ariaLabel?: string;
  analyticsEvent?: AnalyticsEvent;
  onClick?: (event: MouseEvent<HTMLElement>) => void;
}) {
  const localOnClick = (event: MouseEvent<HTMLElement>) => {
    analyticsEvent && logEvent(analyticsEvent);
    onClick && onClick(event);
  };

  return (
    <StyledLink className={className} color={color} href={href} aria-label={ariaLabel} onClick={localOnClick}>
      {children}
    </StyledLink>
  );
}
