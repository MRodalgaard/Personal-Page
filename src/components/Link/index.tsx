import React from 'react';
import styled from 'styled-components';
import { AppColors } from 'util/theme';

const StyledLink = styled.a`
  color: ${(props) => (props.color ? props.color : AppColors.primary)};
  text-decoration: none;
  font-weight: bold;
  transition: color 0.5s ease;

  &:hover {
    color: ${AppColors.secondary};
  }
`;

interface IProps {
  children: React.ReactNode;
  className?: any;
  color?: string;
  href: string;
  ariaLabel?: string;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
}

const Link = ({
  children,
  className,
  color,
  href,
  onClick,
  ariaLabel,
}: IProps) => {
  return (
    <StyledLink
      className={className}
      color={color}
      href={href}
      aria-label={ariaLabel}
      onClick={onClick}
    >
      {children}
    </StyledLink>
  );
};

export default Link;
