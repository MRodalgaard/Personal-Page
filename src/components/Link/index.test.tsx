import { shallow } from 'enzyme';
import React from 'react';
import Link from '.';

describe('Link', () => {
  it('renders', () => {
    const wrapper = shallow(<Link href="#">Mock</Link>);
    expect(wrapper).toMatchSnapshot();
  });
});
