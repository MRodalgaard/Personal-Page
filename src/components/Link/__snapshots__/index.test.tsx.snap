// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `
.c0 {
  color: #000000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  -webkit-transition: color 0.5s ease;
  transition: color 0.5s ease;
}

.c0:hover {
  color: #c9282d;
}

<a
  class="c0"
  href="#"
>
  Mock
</a>
`;
